packages/api/src/web5.ts

```diff
@@ -326,30 +328,37 @@ export class Web5 {
 
       if (registration !== undefined) {
         // If a registration object is passed, we attempt to register the AgentDID and the ConnectedDID with the DWN endpoints provided
-
         const serviceEndpointNodes = techPreview?.dwnEndpoints ?? didCreateOptions?.dwnEndpoints;
-        for (const dwnEndpoint of serviceEndpointNodes) {
-          try {
 
+        // We only want to return the success callback if we successfully register with all DWN endpoints
+        // So we keep track of whether we attempted registration at all
+        let registrationAttempt = false;
+        try {
+          for (const dwnEndpoint of serviceEndpointNodes) {
             // check if endpoint needs registration
             const serverInfo = await userAgent.rpc.getServerInfo(dwnEndpoint);
             if (serverInfo.registrationRequirements.length === 0) {
               // no registration required
               continue;
             }
 
+            registrationAttempt = true;
+
             // register the agent DID
             await DwnRegistrar.registerTenant(dwnEndpoint, agent.agentDid.uri);
 
             // register the connected Identity DID
             await DwnRegistrar.registerTenant(dwnEndpoint, connectedDid);
+          }
 
-            // if no failure occurs, call the onSuccess callback
+          if (registrationAttempt) {
```

Comment:
Is it possible the caller passes in a registration object, and no registration attempt is taken? According to the code path it looks it is possible, but in which case the caller would not be signaled at all regarding a outcome even though they explicitly said they wanted registration to take place, is this okay?